src/Raven.Server/Documents/PeriodicBackup/Aws/RavenAwsS3Client.cs

```diff
@@ -500,6 +495,103 @@ public async Task DeleteBucket()
             }
         }
 
+        public async Task<IEnumerable<FileInfoDetails>> ListObjects(string prefix, string delimiter, bool listFolders, string continuationToken = null)
+        {
+            var url = $"{GetUrl()}/?list-type=2";
+            if (prefix != null)
+                url += $"&prefix={prefix}";
+
+            if (delimiter != null)
+                url += $"&delimiter={delimiter}";
+
+            //url += $"&max-keys={10}";
+
+            if (continuationToken != null)
+                url += $"&continuation-token={Uri.EscapeDataString(continuationToken)}";
+
+            //var k = $"?list-type=2&prefix={prefix}";
+
+            var now = SystemTime.UtcNow;
+
+            var requestMessage = new HttpRequestMessage(HttpMethods.Get, url);
+            UpdateHeaders(requestMessage.Headers, now, null);
+
+            var headers = ConvertToHeaders(requestMessage.Headers);
+
+            var client = GetClient();
+            client.DefaultRequestHeaders.Authorization = CalculateAuthorizationHeaderValue(HttpMethods.Get, url, now, headers);
+
+            var response = await client.SendAsync(requestMessage, CancellationToken);
+            if (response.StatusCode == HttpStatusCode.NotFound)
+                return new List<FileInfoDetails>();
+
+            if (response.IsSuccessStatusCode == false)
+                throw StorageException.FromResponseMessage(response);
+
+            var responseStream = await response.Content.ReadAsStreamAsync();
+            var listBucketResult = XDocument.Load(responseStream);
```

Comment:
Is there a JSON endpoint available?

Reply:
No. Only XML.
https://docs.aws.amazon.com/AmazonS3/latest/dev/ListingObjectKeysUsingREST.html